deployment/terraform/resources/storage_account.tf

```diff
@@ -53,3 +76,18 @@ resource "azurerm_role_assignment" "pctasks-server-table-access" {
   role_definition_name = "Storage Table Data Contributor"
   principal_id         = var.pctasks_server_sp_object_id
 }
+
+# Let the Azure Functions in pctasks process queue messages.
+resource "azurerm_role_assignment" "pctasks-functions-queue-access" {
+  scope                = azurerm_storage_account.pctasks.id
+  role_definition_name = "Storage Queue Data Contributor"
+  principal_id         = azurerm_linux_function_app.pctasks.identity[0].principal_id
+}
+
+# The Task Service Principal should be able to process queue messages
+# for the dataset work queues
+resource "azurerm_role_assignment" "pctasks-task-queue-access" {
```

Comment:
This role assignment lets the Task service principal process messages from *every* queue, including "management" queues that pctasks uses internally. I think we aren't using anything on `main`, but in this PR we're using a `storage-events` and `ingest` queue.

If that's undesirable, we *could* do another `for_each` here and only grant access to the dataset work queues. Ah, but we'll also need to add the `ingest` queue, since the task Service Principal will process messages from it. So it'd really just be protecting the `storage-events` queue (which receives the storage events from event grid subscriptions).